lib/util.js

```diff
@@ -19,22 +20,42 @@ const requestDefaults = {
         'User-Agent': USER_AGENT,
     },
 };
-let pooledRequest = request.defaults(requestDefaults);
+
+// Exponential wait: 800ms, 1600ms, 3200ms
+const MAX_RETRIES = 3;
+const RETRY_START_MS = 800;
 
 // Wrap request results in a promise
-function rp (options) {
-    return new Promise((resolve, reject) => {
-        pooledRequest(options, (err, res, body) => {
-            if (err) return reject(err);
-            resolve(res);
-        });
+let rp = promisify(request.defaults(requestDefaults));
+
+function wait (timeout) {
+    return new Promise((resolve) => {
+        setTimeout(() => {
+            resolve()
+        }, timeout);
     });
 }
 
+
+async function requestRetrying (options) {
+    for (let i = 0; i < MAX_RETRIES; i++) {
+        try {
+            return await rp(options);
+        } catch (err) {
+            if (err.code == 'ECONNRESET' || err.code == 'ETIMEDOUT') {
+                const timeout = Math.pow(2, i) * RETRY_START_MS;
+                await wait(timeout);
```

Comment:
Ditto on await

Reply:
Given that we don't have await, maybe this needs to be done with promise chains rather than loop too